app/components-react/root/LiveDock.tsx

```diff
@@ -0,0 +1,437 @@
+
+class LiveDockController {
+  private streamingService = Services.StreamingService;
+  private youtubeService = Services.YoutubeService;
+  private facebookService = Services.FacebookService;
+  private trovoService = Services.TrovoService;
+  private userService = Services.UserService;
+  private customizationService = Services.CustomizationService;
+  private platformAppsService = Services.PlatformAppsService;
+  private appService = Services.AppService;
+  private chatService = Services.ChatService;
+  private windowsService = Services.WindowsService;
+  private restreamService = Services.RestreamService;
+
+  store = initStore({
+    elapsedStreamTime: '',
+    canAnimate: false,
+    slot: EAppPageSlot.Chat,
+    underlyingSelectedChat: 'default',
+  });
+
+  // Safe getter/setter prevents getting stuck on the chat
+  // for an app that was unloaded.
+  setChat(key: string) {
+    this.store.setState(s => (s.underlyingSelectedChat = key));
+  }
+
+  get selectedChat() {
+    if (this.store.underlyingSelectedChat === 'default') return 'default';
+    if (this.store.underlyingSelectedChat === 'restream') {
+      if (this.restreamService.shouldGoLiveWithRestream) return 'restream';
+      return 'default';
+    }
+
+    return this.chatApps.find(app => {
+      if (app.id === this.store.underlyingSelectedChat) {
+        return app.id === this.store.underlyingSelectedChat;
+      }
+    })
+      ? this.store.underlyingSelectedChat
+      : 'default';
+  }
+
+  get applicationLoading() {
+    return this.appService.state.loading;
+  }
+
+  get streamingStatus() {
+    return this.streamingService.state.streamingStatus;
+  }
+
+  get isStreaming() {
+    return this.streamingService.isStreaming;
+  }
+
+  get collapsed() {
+    return this.customizationService.state.livedockCollapsed;
+  }
+
+  get liveText() {
+    if (this.streamingStatus === EStreamingState.Live) return 'Live';
+    if (this.streamingStatus === EStreamingState.Starting) return 'Starting';
+    if (this.streamingStatus === EStreamingState.Ending) return 'Ending';
+    if (this.streamingStatus === EStreamingState.Reconnecting) return 'Reconnecting';
+    return 'Offline';
+  }
+
+  get platform() {
+    return this.userService.platform?.type;
+  }
+
+  get offlineImageSrc() {
+    const mode = this.customizationService.isDarkTheme ? 'night' : 'day';
+    return require(`../../../media/images/sleeping-kevin-${mode}.png`);
+  }
+
+  get hideViewerCount() {
+    return this.customizationService.state.hideViewerCount;
+  }
+
+  get liveDockSize() {
+    return this.customizationService.state.livedockSize;
+  }
+
+  get viewerCount() {
+    if (this.hideViewerCount) {
+      return 'viewers hidden';
+    }
+    return this.streamingService.views.viewerCount.toString();
+  }
+
+  get hideStyleBlockers() {
+    return this.windowsService.state.main.hideStyleBlockers;
+  }
+
+  get hasChatTabs() {
+    return this.chatTabs.length > 1;
+  }
+
+  get showDefaultPlatformChat() {
```

Comment:
Cosmetic: perhaps rename to `shouldShowDefaultPlatformChat` as it's not an action?